SpatialGDK/Source/SpatialGDK/Public/EngineClasses/SpatialServerWorkerSystem.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "SpatialCommonTypes.h"
+#include "SpatialView/ComponentData.h"
+#include "Subsystems/GameInstanceSubsystem.h"
+
+#include "SpatialServerWorkerSystem.generated.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogSpatialServerWorkerSystem, Log, All);
+
+namespace SpatialGDK
+{
+class FServerWorkerSystemImpl;
+} // namespace SpatialGDK
+
+class USpatialNetDriver;
+
+UCLASS(Abstract)
+class SPATIALGDK_API USpatialServerWorkerSystem : public UGameInstanceSubsystem
+{
+	GENERATED_BODY()
+
+public:
+	USpatialServerWorkerSystem();
+	USpatialServerWorkerSystem(FVTableHelper&);
+	~USpatialServerWorkerSystem();
+
+	virtual bool ShouldCreateSubsystem(UObject* Outer) const override;
+
+	virtual TArray<SpatialGDK::ComponentData> GetServerWorkerData();
```

Comment:
Minor point - I couldn't tell from the name if this is meant called by a user to get the data in the view or if this is called by internet systems to get initial data (which is actually what it seems to do). Would be good to reflect that in the name.

Reply:
The fact that it was virtual was a hint in fairness but it's also a public function rather than a private virtual one that gets called as part of initialisation.

Reply:
Refactored it and made it private.